plugin/mutate/source/dextool/plugin/mutate/backend/report/html/page_tree_map.d

```diff
@@ -0,0 +1,100 @@
+module dextool.plugin.mutate.backend.report.html.page_tree_map;
+
+import std.json : JSONValue;
+import arsd.dom : Document, Element, require, Table, RawSource;
+
+import dextool.plugin.mutate.backend.report.html : FileIndex;
+import dextool.plugin.mutate.backend.report.html.tmpl : tmplBasicPage;
+
+/* 
+* A JSON-like object that is more easily manipulated than JSONValue.
+* Has operator overloads for "index", "index assign" and "in" allowing the children list to be private
+* and cleaner construction of the object. 
+*/
+class JSONLike {
+    string name;
+    private JSONLike[string] children;
+    long locs = -1;
```

Comment:
Tip: Instead of a "magic value" that signals a behaviour you can use `Nullable!long`. It default initializes to "null".
Then it is possible at line 30 to check if it contains a value via `.isNull` and if so access the value via `.get`.